fe/fe-core/src/main/java/com/starrocks/qe/scheduler/dag/ExecutionFragment.java

```diff
@@ -167,10 +168,19 @@ public ColocatedBackendSelector.Assignment getColocatedAssignment() {
         return colocatedAssignment;
     }
 
-    public ColocatedBackendSelector.Assignment getOrCreateColocatedAssignment(OlapScanNode scanNode) {
+    public ColocatedBackendSelector.Assignment getOrCreateColocatedAssignment(ScanNode scanNode)
+            throws StarRocksException {
         if (colocatedAssignment == null) {
-            final int numOlapScanNodes = scanNodes.values().stream().mapToInt(node -> node instanceof OlapScanNode ? 1 : 0).sum();
-            colocatedAssignment = new ColocatedBackendSelector.Assignment(scanNode, numOlapScanNodes);
+            final int numScanNodes = scanNodes.size();
+            ColocatedBackendSelector.Assignment.ScanRangeType type;
+            if (scanNode instanceof OlapScanNode olapScanNode) {
```

Comment:
What is the variable olapScanNode used for?

Reply:
in the original implementation, olapscannode was used to calc bucket num and now I add a getBucketNums interface, and use bucket num as the parameter of ColocatedBackendSelector.Assignment.